src/backend/dx11/src/lib.rs

```diff
@@ -997,19 +1100,50 @@ impl hal::command::RawCommandBuffer<Backend> for CommandBuffer {
         // unimplemented!()
     }
 
-    fn clear_image<T>(&mut self, image: &Image, _: image::Layout, color: command::ClearColorRaw, _depth_stencil: command::ClearDepthStencilRaw, subresource_ranges: T)
+    fn clear_image<T>(&mut self, image: &Image, _: image::Layout, color: command::ClearColorRaw, depth_stencil: command::ClearDepthStencilRaw, subresource_ranges: T)
     where
         T: IntoIterator,
         T::Item: Borrow<image::SubresourceRange>,
     {
-        // TODO: use a internal program to clear for subregions in the image
-        for subresource_range in subresource_ranges {
-            let _sub = subresource_range.borrow();
-            unsafe {
-                self.context.ClearRenderTargetView(
-                    image.get_rtv(0, 0).unwrap().as_raw(),
-                    &color.float32
-                );
+        for range in subresource_ranges {
+            let range = range.borrow();
+
+            if range.aspects.contains(format::Aspects::COLOR) {
+                for layer in range.layers.clone() {
+                    for level in range.levels.clone() {
+                        unsafe {
+                            self.context.ClearRenderTargetView(
+                                image.get_rtv(level, layer).unwrap().as_raw(),
+                                &color.float32
+                            );
+                        }
+                    }
+                }
+            }
+
+            // TODO: clear foreach level, layer on DSVs
```

Comment:
aren't you iterating the levels/layers already?

Reply:
yup, heat must be getting to me 😓